db/internal_stats.cc

```diff
@@ -1104,8 +1193,18 @@ bool InternalStats::HandleEstimateTableReadersMem(uint64_t* value,
 
 bool InternalStats::HandleEstimateLiveDataSize(uint64_t* value, DBImpl* /*db*/,
                                                Version* version) {
+  uint64_t estimateNonBlobDataSize = 0;
   const auto* vstorage = version->storage_info();
-  *value = vstorage->EstimateLiveDataSize();
+  estimateNonBlobDataSize = vstorage->EstimateLiveDataSize();
+
+  uint64_t estimateBlobDataSize = 0;
+  const auto& blobFiles = vstorage->GetBlobFiles();
+  for (const auto& pair : blobFiles) {
+    const auto& meta = pair.second;
+    estimateBlobDataSize += meta->GetTotalBlobBytes();
+    estimateBlobDataSize -= meta->GetGarbageBlobBytes();
+  }
+  *value = estimateNonBlobDataSize + estimateBlobDataSize;
```

Comment:
Similarly, this part could be folded into `VersionStorageInfo::EstimateLiveDataSize`